src/main/java/com/tterrag/registrate/AbstractRegistrate.java

```diff
@@ -582,38 +677,60 @@ public S object(String name) {
         return self();
     }
 
-    /**
-     * Set the default creative mode tab for all future items created with this Registrate, until the next time this method is called. The supplier will only be called once, and the value re-used for each
-     * entry.
-     *
-     * @param tab
-     *            The tab to use for future items
-     * @return this {@link AbstractRegistrate}
-     */
-    @ApiStatus.ScheduledForRemoval(inVersion = "1.20")
-    @Deprecated(forRemoval = true, since = "1.19.3")
-    public S creativeModeTab(NonNullSupplier<? extends CreativeModeTab> tab) {
-        this.currentTab = Suppliers.memoize(tab::get);
+    public Supplier<CreativeModeTab> buildCreativeModeTab(String registryName, @Nullable List<Object> beforeEntries, @Nullable List<Object> afterEntries, Consumer<CreativeModeTab.Builder> configurator, @Nullable String englishTranslation) {
```

Comment:
It's going to be a lot of boilerplate unfortunately, but these methods need some javadoc. Especially explaining the List<object> params. I'm reading the vanilla code that uses them and I'm still not 100% sure how they work...